cpmpy/solvers/pysat_rc2.py

```diff
@@ -0,0 +1,172 @@
+            - time_limit:  maximum solve time in seconds (float, optional). Auto-interrups in case the
+                           runtime exceeds given time_limit.
+                           Warning: the time_limit is not very accurate at subsecond level
+            - assumptions: list of CPMpy Boolean variables that are assumed to be true.
+                           For use with s.get_core(): if the model is UNSAT, get_core() returns a small subset of assumption variables that are unsat together.
+                           Note: the PySAT interface is statefull, so you can incrementally call solve() with assumptions and it will reuse learned clauses
+        """
+        ## TODO: adding assumptions to the solving !
+        if assumptions is not None:
+            print("[Warning] Assumption variables interpreted as hard clauses!")
+            pysat_assum_vars = self.solver_vars(assumptions)
+            for pysat_assum_var in pysat_assum_vars:
+                self.pysat_solver.add_clause([pysat_assum_var])
+
+        sol = self.pysat_solver.compute()
+
+        # new status, translate runtime
+        self.cpm_status = SolverStatus(self.name)
+        self.cpm_status.runtime = self.pysat_solver.oracle_time()
+
+        # translate exit status
+        if sol is not None:
+            self.cpm_status.exitstatus = ExitStatus.FEASIBLE
+        elif sol is None:
+            self.cpm_status.exitstatus = ExitStatus.UNSATISFIABLE
+
+        # True/False depending on self.cpm_status
+        has_sol = self._solve_return(self.cpm_status)
+
+        # translate solution values (of user specified variables only)
+        if has_sol:
+            sol = frozenset(sol)  # to speed up lookup
+            # fill in variable values
+            for cpm_var in self.user_vars:
+                lit = self.solver_var(cpm_var)
+                if lit in sol:
+                    cpm_var._value = True
+                elif -lit in sol:
+                    cpm_var._value = False
+                else:
+                    # not specified...
+                    cpm_var._value = None
+
+        return has_sol
+
+    def objective(self, expr, minimize):
+        if minimize:
+            raise NotImplementedError()
+
+        if isinstance(expr, Operator) and expr.name == "sum":
+            pysat_assum_vars = self.solver_vars(expr.args)
+            for pysat_assum_var in pysat_assum_vars:
+                self.pysat_solver.add_clause([pysat_assum_var], weight=1)
+        elif isinstance(expr, Operator) and expr.name == "wsum":
+            weights, vars = expr.args
+            pysat_assum_vars = self.solver_vars(vars)
+            for weight, pysat_assum_var in zip(weights, pysat_assum_vars):
+                self.pysat_solver.add_clause([pysat_assum_var], weight=weight)
+        else:
+            raise NotImplementedError(f"Expression {expr} not handled")
+
+    def get_core(self):
+        raise NotImplementedError("Does not work.")
```

Comment:
I think the error message should be a little more elaborate :) Something like "RC2 does not support unsat core extraction, check out the PySat solver for this functionality..."